[2024-07-15T20:09:33.812+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-07-15T20:09:33.839+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: csv_to_pgsql.extract_and_load_process manual__2024-07-15T20:09:23.760382+00:00 [queued]>
[2024-07-15T20:09:33.846+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: csv_to_pgsql.extract_and_load_process manual__2024-07-15T20:09:23.760382+00:00 [queued]>
[2024-07-15T20:09:33.847+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2024-07-15T20:09:33.859+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): extract_and_load_process> on 2024-07-15 20:09:23.760382+00:00
[2024-07-15T20:09:33.865+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'csv_to_pgsql', 'extract_and_load_process', 'manual__2024-07-15T20:09:23.760382+00:00', '--job-id', '270', '--raw', '--subdir', 'DAGS_FOLDER/from_csv_to_pgsql.py', '--cfg-path', '/tmp/tmp75e0q4o4']
[2024-07-15T20:09:33.867+0000] {standard_task_runner.py:91} INFO - Job 270: Subtask extract_and_load_process
[2024-07-15T20:09:33.867+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=1717) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-07-15T20:09:33.868+0000] {standard_task_runner.py:63} INFO - Started process 1718 to run task
[2024-07-15T20:09:33.907+0000] {task_command.py:426} INFO - Running <TaskInstance: csv_to_pgsql.extract_and_load_process manual__2024-07-15T20:09:23.760382+00:00 [running]> on host 1ee09c187a7a
[2024-07-15T20:09:34.001+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='csv_to_pgsql' AIRFLOW_CTX_TASK_ID='extract_and_load_process' AIRFLOW_CTX_EXECUTION_DATE='2024-07-15T20:09:23.760382+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-07-15T20:09:23.760382+00:00'
[2024-07-15T20:09:34.003+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-07-15T20:09:50.619+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-07-15T20:09:50.620+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/from_csv_to_pgsql.py", line 27, in extract_and_load_task
    df.write.jdbc(url=postgres_uri, table=f'ds.{file_name}', mode="overwrite", properties=properties)
    ^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/core/generic.py", line 6204, in __getattr__
    return object.__getattribute__(self, name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'DataFrame' object has no attribute 'write'
[2024-07-15T20:09:50.639+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=csv_to_pgsql, task_id=extract_and_load_process, run_id=manual__2024-07-15T20:09:23.760382+00:00, execution_date=20240715T200923, start_date=20240715T200933, end_date=20240715T200950
[2024-07-15T20:09:50.664+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 270 for task extract_and_load_process ('DataFrame' object has no attribute 'write'; 1718)
[2024-07-15T20:09:50.706+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-07-15T20:09:50.739+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-07-15T20:09:50.744+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
